src/components/views/LocalModList/SearchAndSort.vue

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { DeferredInput } from '../../all';
+import GameManager from '../../../model/game/GameManager';
+import { SortDirection } from '../../../model/real_enums/sort/SortDirection';
+import { SortLocalDisabledMods } from '../../../model/real_enums/sort/SortLocalDisabledMods';
+import { SortNaming } from '../../../model/real_enums/sort/SortNaming';
+import ManagerSettings from '../../../r2mm/manager/ManagerSettings';
+
+@Component({
+    components: {
+        DeferredInput,
+    }
+})
+export default class SearchAndSort extends Vue {
+    settings = new ManagerSettings();
+
+    get order() {
+        return this.$store.state.profile.order;
+    }
+
+    set order(value: SortNaming) {
+        this.$store.commit('profile/setOrder', value);
+        this.settings.setInstalledSortBy(value);
+    }
+
+    get direction() {
+        return this.$store.state.profile.direction;
+    }
+
+    set direction(value: SortDirection) {
+        this.$store.commit('profile/setDirection', value);
+        this.settings.setInstalledSortDirection(value);
+    }
+
+    get disabledPosition() {
+        return this.$store.state.profile.disabledPosition;
+    }
+
+    set disabledPosition(value: SortLocalDisabledMods) {
+        this.$store.commit('profile/setDisabledPosition', value);
+        this.settings.setInstalledDisablePosition(value);
+    }
+
+    get search() {
+        return this.$store.state.profile.searchQuery;
+    }
+
+    set search(value: string) {
+        this.$store.commit('profile/setSearchQuery', value);
+    }
+
+    get orderOptions() {
+        return Object.values(SortNaming);
+    }
+
+    get directionOptions() {
+        return Object.values(SortDirection);
+    }
+
+    get disabledOptions() {
+        return Object.values(SortLocalDisabledMods);
+    }
+
+    async created() {
+        this.settings = await ManagerSettings.getSingleton(GameManager.activeGame);
+
+        this.$store.commit('profile/initialize', [
+            this.settings.getInstalledSortBy(),
+            this.settings.getInstalledSortDirection(),
+            this.settings.getInstalledDisablePosition()
+        ]);
+    }
+
+    async destroyed() {
+        this.$store.commit('profile/setSearchQuery', '');
```

Comment:
Added this so the search query gets cleared when the component is unmounted, to match the existing functionality.